### PR TITLE
GitHub: Allow operators to set entity ingestion namespace

### PR DESCRIPTION
**Pull request by mspiegel31**
## Hey, I just made a Pull Request!
This PR modifies the configuration for `GithubEntityProvider`.  It allows an operator to configure an optional `namespace` that ingested locations are written to.

### Sample usecase
Your company has acquired another company (hooray!).  This company has a github org you'd like to add to backstage.  However, several repositories between the organizations have the exact same name.  This will result in the `conflictingEntityRef` message if `orgA/Repo` and `orgB/Repo` are both added to the `default` namespace.  This PR aims to solve that

### Outstanding issues
I couldn't find the spot in the code where locations are fetched, only written 😓 .  I'm not sure if `Location` namespaces are respected/copied to their corresponding `Component`/`Resource`/`Whatever` Entities.

If there's a better place to submit this modification, please let me know!

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] (Not applicable) Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-backend-module-github/src/providers/GithubEntityProviderConfig.ts [hunk 1]
```diff
@@ -69,6 +71,7 @@ function readProviderConfig(
   config: Config,
 ): GithubEntityProviderConfig {
   const organization = config.getString('organization');
+  const namespace = config.getOptionalString('namespace');
```
Comment (Rugvip) [inline plugins/catalog-backend-module-github/src/providers/GithubEntityProviderConfig.ts:74]: Best default it up here while reading, like the rest

File: docs/integrations/github/discovery.md [hunk 2]
```diff
@@ -173,6 +173,8 @@ If you do so, `default` will be used as provider ID.
 - **`organization`**:
   Name of your organization account/workspace.
   If you want to add multiple organizations, you need to add one provider config each.
+- **`namespace`** _(optional)_:
+  The namespace to assign to ingested entities. Uses the `default` namespace if not provided.
```
Comment (Rugvip) [inline docs/integrations/github/discovery.md:177]: Best to highlight that this only applies to the location entities right? Not the actual entities themselves

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! :+1: 

Couple of nits

File: docs/integrations/github/discovery.md (hunk 2 above)
Comment (mspiegel31) [reply docs/integrations/github/discovery.md:177]: @Rugvip ah, that's sort of my question;  My goal here is to make sure the _generated_ entities (`Components` etc) are added to this namespace.  Is this the right spot for modification?  as I mentioned in the PR, I couldn't find the place in the processing/provdier loop where locations were fetched.

Comment (mspiegel31) [reply docs/integrations/github/discovery.md:177]: @Rugvip after some digging this weekend, I think that #25814 handles this problem more gracefully.  I'm closing this one in favor of that.  Thanks for your feedback thus far!

**Comment by mspiegel31** [comment]
@Rugvip after some digging this weekend, I think that #25814 handles this problem more gracefully.  I'm closing this one in favor of that.  Thanks for your feedback thus far!

**Pull request closed without merging**